packages/playwright-test/src/plugins/gitCommitInfoPlugin.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { Config, Metadata, TestPlugin } from '../types';
+import { createGuid } from 'playwright-core/lib/utils';
+import { spawnAsync } from 'playwright-core/lib/utils/spawnAsync';
+
+const GIT_OPERATIONS_TIMEOUT_MS = 1500;
+
+export const gitCommitInfo = (options?: GitCommitInfoPluginOptions): TestPlugin => {
+  return {
+    name: 'playwright-git-commit-info-plugin',
+
+    configure: async (config: Config, configDir: string) => {
+      const info = {
+        ...linksFromEnv(),
+        ...options?.info ? options.info : await gitStatusFromCLI(options?.directory || configDir),
+        generatedAt: Date.now(),
+      };
+        // Normalize dates
+      const timestamp = info['revision.timestamp'];
+      if (timestamp instanceof Date)
+        info['revision.timestamp'] = timestamp.getTime();
+
+      config.metadata = config.metadata || {};
+      config.metadata['git-commit-info'] = info as Metadata;
```

Comment:
Yeah, so I was assuming we are using global namespace for revisions. I.e. metadata['revision.timestamp'] so that non-git could use the same namespace.